src/app/plugins/plugins.module.ts

```diff
@@ -0,0 +1,16 @@
+import { NgModule } from '@angular/core';
+
+import { DemoComponent } from './demo.component';
+import { PluginDirective } from './plugin.directive';
+import { PluginsService } from './plugins.service';
+
+@NgModule({
+    imports: [],
+    exports: [PluginDirective],
+    declarations: [
+        DemoComponent,
+        PluginDirective
+    ],
+    providers: [PluginsService],
+})
+export class PluginsModule { }
```

Comment:
I was thinking in context of applications, where a module extends the capabilities of the application.